channeldb/graph.go

```diff
@@ -1198,28 +1208,46 @@ func fetchLightningNode(nodeBucket *bolt.Bucket,
 
 func deserializeLightningNode(r io.Reader) (*LightningNode, error) {
 	node := &LightningNode{}
-
 	var scratch [8]byte
+
 	if _, err := r.Read(scratch[:]); err != nil {
 		return nil, err
 	}
 
 	unix := int64(byteOrder.Uint64(scratch[:]))
 	node.LastUpdate = time.Unix(unix, 0)
 
-	addrString, err := wire.ReadVarString(r, 0)
-	if err != nil {
+	if _, err := r.Read(scratch[:2]); err != nil {
 		return nil, err
 	}
-	node.Address, err = net.ResolveTCPAddr("tcp", addrString)
-	if err != nil {
-		return nil, err
+	numAddresses := int(byteOrder.Uint16(scratch[:2]))
+
+	addresses := make([]net.Addr, 0)
+	for i := 0; i < numAddresses; i++ {
+		network, err := wire.ReadVarString(r, 0)
+		if err != nil {
+			return nil, err
+		}
+
+		addr, err := wire.ReadVarString(r, 0)
+		if err != nil {
+			return nil, err
+		}
+
+		address, err := net.ResolveTCPAddr(network, addr)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
In light of my comment above, I think we should instead add a single byte prefix before the encoded address so we know how to properly decode it. 

Also the address resolution above wouldn't work with say, `.onion` addresses.